src/cat.jl

```diff
@@ -18,64 +22,87 @@ struct ConcatDiskArray{T,N,P,C,HC} <: AbstractDiskArray{T,N}
     chunks::C
     haschunks::HC
 end
-function ConcatDiskArray(arrays::AbstractArray{<:AbstractArray{<:Any,N},M}) where {N,M}
-    T = mapreduce(eltype, promote_type, init=eltype(first(arrays)), arrays)
 
-    function othersize(x, id)
-        return (x[1:(id-1)]..., x[(id+1):end]...)
+function ConcatDiskArray(arrays::AbstractArray{Union{<:AbstractArray,Missing}})
+    et = Base.nonmissingtype(eltype(arrays))
+    T = Union{Missing,eltype(et)}
```

Comment:
```suggestion
    T = promotetype(typeof(fill), eltype(et))
```
Would be good to all arbitrary fill values, like 0.0